kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -221,12 +221,8 @@ public Transaction build(Engine engine) {
       newMetadata = icebergCompatV2Metadata;
     }
 
-    Optional<Metadata> universalFormatMetadata =
-        IcebergUniversalFormatMetadataValidatorAndUpdater.validateAndUpdate(
-            snapshotMetadata, newMetadata.orElse(snapshotMetadata));
-    if (universalFormatMetadata.isPresent()) {
-      newMetadata = universalFormatMetadata;
-    }
+    IcebergUniversalFormatMetadataValidatorAndUpdater.validate(
+        newMetadata.orElse(snapshotMetadata));
```

Comment:
Oh awesome :) if we're only validating what do you think about putting it in `validateMetadataChange` helper?

Reply:
done.